### PR TITLE
Fix crash: Skip unused value_info for version_converter

### PR DESCRIPTION
**Pull request by jcwchen**
**Description**
- Add a common CreateDummyValue function for version_converter
- ~Create dummy value for value_info whose name does not exist in the graph. Or, perhaps it should throw an error or warning since it might be an invalid model~. If a value_info's name does not exist in the graph (unused), simply skip it during version conversion.

**Motivation and Context**
Closes https://github.com/onnx/onnx/issues/4078.

### CODE REVIEW

**Review by liqunfu — APPROVED** [review]

**Comment by gramalingam** [comment]
I am unclear about the exact invariants of the IR and the design. Would it be possible to add a line or so near https://github.com/onnx/onnx/blob/94e2f64551ded652df53a7e9111031e8aabddaee/onnx/common/ir_pb_converter.cc#L223 to at least document the creation of dummy nodes for valueinfos without a definition?

File: onnx/common/ir_pb_converter.cc [hunk 1]
```diff
@@ -204,6 +204,13 @@ std::vector<Dimension> tensorShapeProtoToDimensions(const ONNX_NAMESPACE::Tensor
   return dims;
 }
 
+void createDummyValue(std::unique_ptr<Graph>& g, const std::string& name, std::unordered_map<std::string, Value*>& value_by_name_of) {
+  auto* undef = g->create(kCaptured, 1);
```
Comment (gramalingam) [inline onnx/common/ir_pb_converter.cc:208]: What exactly does `kCaptured` mean? Maybe it is relevant to the other two original calls, but not sure it means the same for undefined valueinfos.

**Comment by jcwchen** [comment]
After discussion, we think simply skipping such a unused value_info should be fine for now. Then we don't need to create a dummy node for those unused value_infos.

File: onnx/common/ir_pb_converter.cc (hunk 1 above)
Comment (jcwchen) [reply onnx/common/ir_pb_converter.cc:208]: Good question. It seems to me that `kCaptured` is used for an inner (nested) graph that some inputs/outputs might not be visible in value_by_name_of so it creates a dummy node for them with `kCaptured` type.

**Review by gramalingam — APPROVED** [review]

**Pull request merged**